Makefile

```diff
@@ -504,10 +504,14 @@ promtail-push: promtail-image-cross
 	$(call push-image,promtail)
 
 # loki
+# Set docker platform linux/amd64 for arm64 processors architecture
```

Comment:
This is really great. Nice job tracking down the bug @ssncferreira :)

This change would _never_ allow to build Loki images for arm architectures _even if we want to_ correct?
- say I run k8 cluster on raspberry pi nodes
- I want to _run_ the docker image on Apple M1.

How about we make `amd64` as default and let users allow to set it to `arm64` (or any other) if they want to?

I _suggest_ following changes.

1. Instead of setting `BUILD_PLATFORM` conditionally, may be make it as default to `amd64`? and able to change it if anyone want's to

e.g: Instead of this
```
ifeq ($(shell uname -p),arm)
BUILD_PLATFORM=--platform=linux/amd64
endif
```
how about just this?
```
BUILD_PLATFORM ?= amd64 
```
NOTE: `?=` make it easy to set the custom value for that ENV. And we don't need `linux` prefix in platform as `loki-image-cross` is already only for linux?

and change `loki-image-cross` accordingly.

```
$(SUDO) $(BUILD_OCI) --platform=linux/$(BUILD_PLATFORM) -t $(IMAGE_PREFIX)/loki:$(IMAGE_TAG) -f cmd/loki/Dockerfile.cross .
```

2. Add some info message while building `loki-image-cross`?

Something like 
```
make loki-image-cross build docker image for amd64 architecture by default, 
if you want to build for different architecture, say for arm64 (Raspberry Pi or Apple Silicon),
set BUILD_PlATFORM=arm64.
```

Reply:
Thanks @kavirajk yes that makes absolute sense 🏆 
Addressed in commit https://github.com/grafana/loki/pull/4981/commits/c1bc934a0576118216d892258c693cc8627854bf